ui/guest/src/controls/rte.ts

```diff
@@ -114,6 +116,42 @@ export function initTinyMCE(
     external_plugins: external,
     code_editor_inline: false,
     skin: window.matchMedia('(prefers-color-scheme: dark)').matches ? 'oxide-dark' : 'oxide',
+
+    media_live_embeds: true,
+    file_picker_types: 'image media',
+    file_picker_callback: function (cb, value, meta) {
+      // meta contains info about type (image, media, etc). Used to properly add DS to dialogs.
+      // meta.filetype === 'file | image | media'
+      const datasources = {};
+      Object.values(field.validations).forEach((validation) => {
+        if (
+          ['allowImageUpload', 'allowImagesFromRepo', 'allowVideoUpload', 'allowVideosFromRepo'].includes(validation.id)
+        ) {
+          datasources[validation.id] = validation;
+        }
+      });
+      const browseBtn = document.querySelector('.tox-dialog .tox-browse-url');
+
+      post({
+        type: showRtePickerActions.type,
+        payload: {
+          datasources,
+          model,
+          type: meta.filetype,
+          btnRect: browseBtn.getBoundingClientRect()
+        }
+      });
+
+      message$
+        .pipe(
+          filter((e) => e.type === rtePickerActionResult.type),
```

Comment:
Make sure you're sending both ok and cancel actions so this observable dies with either result.